projects/igniteui-angular/src/lib/grids/filtering/base/grid-filtering-row.component.html

```diff
@@ -122,6 +122,20 @@
 </button>
 
 <div #buttonsContainer class="igx-grid__filtering-row-editing-buttons">
-    <button igxButton igxRipple (click)="clearFiltering()" [disabled]="disabled" [tabindex]="disabled">{{filteringService.grid.resourceStrings.igx_grid_filter_row_reset}}</button>
-    <button #closeButton igxButton igxRipple (click)="close()">{{filteringService.grid.resourceStrings.igx_grid_filter_row_close}}</button>
+    <button igxButton="flat" igxRipple (click)="clearFiltering()" [disabled]="disabled" [tabindex]="disabled">
+        <igx-icon>refresh</igx-icon>
+        {{filteringService.grid.resourceStrings.igx_grid_filter_row_reset}}
+    </button>
+    <button #closeButton igxButton="flat" igxRipple (click)="close()">
+        <igx-icon>close</igx-icon>
+        {{filteringService.grid.resourceStrings.igx_grid_filter_row_close}}
+    </button>
+
+    <!-- Visible only under 432px viewport  -->
+    <button igxButton="icon" igxRipple (click)="clearFiltering()" [disabled]="disabled" [tabindex]="disabled">
+        <igx-icon>refresh</igx-icon>
+    </button>
+    <button #closeButton igxButton="icon" igxRipple (click)="close()">
+        <igx-icon>close</igx-icon>
+    </button>
```

Comment:
Maybe instead of having 4 buttons, we could just have the original two, with the icons hidden in desktop mode with only the text visible, and vice-versa in the mobile view.

Reply:
I was thinking the same thing, but then I had to modify the mobile styles of the flat button in the context of filtering row to look like an icon button.